fake_joint_driver/src/fake_joint_driver.cpp

```diff
@@ -96,20 +114,28 @@ FakeJointDriver::FakeJointDriver(void)
     }
   }
 
-  // Create joint_state_interface and position_joint_interface
-  for (int i = 0; i< joint_names_.size(); i++)
+  // Create joint_state_interface
+  for (int i = 0; i < joint_names_.size(); i++)
   {
-    ROS_DEBUG_STREAM("joint[" << i << "]:" << joint_names_[i]);
+    RCLCPP_DEBUG_STREAM(LOGGER, "joint[" << i << "]:" << joint_names_[i]);
     // Connect and register the joint_state_interface
-    hardware_interface::JointStateHandle state_handle(joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
-    joint_state_interface.registerHandle(state_handle);
+    joint_state_handles_[i] =
+        hardware_interface::JointStateHandle(joint_names_[i], &act_dis[i], &act_vel[i], &act_eff[i]);
+    if (register_joint_state_handle(&joint_state_handles_[i]) != hardware_interface::HW_RET_OK)
+      throw std::runtime_error("unable to register " + joint_state_handles_[i].get_name());
```

Comment:
maybe better use ROS_FATAL for this. I don't think the driver needs to throw exceptions